### PR TITLE
RBAC: remove RBAC enabled for Alerting

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
- Adds correct permissions for signedIsUsers for performing view/writes to folder/dashboards
- 

**Which issue(s) does this PR fix?**:

Fixes #
https://github.com/grafana/grafana-enterprise/issues/4942

### CODE REVIEW

File: pkg/services/alerting/store.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func (ss *sqlStore) HandleAlertsQuery(ctx context.Context, query *alertmodels.Ge
 			builder.Write(")")
 		}
 
+		// FIXME: should not read for orgroles but only permissions
 		if query.User.OrgRole != org.RoleAdmin {
```
Comment (eleijonmarck) [inline pkg/services/alerting/store.go:172]: @IevaVasiljeva we should refactor this right? as we should not check for Roles

File: pkg/services/alerting/store_test.go [hunk 2]
```diff
@@ -49,12 +50,12 @@ func TestIntegrationAlertingDataAccess(t *testing.T) {
 		ss := db.InitTestDB(t)
 		tagService := tagimpl.ProvideService(ss, ss.Cfg)
 		cfg := setting.NewCfg()
-		cfg.RBACEnabled = false
 		store = &sqlStore{
 			db:         ss,
 			log:        log.New(),
 			cfg:        cfg,
 			tagService: tagService,
+			features:   featuremgmt.WithFeatures(),
```
Comment (eleijonmarck) [inline pkg/services/alerting/store_test.go:58]: needed for not getting a nil pointer exception in the Permissions filtering, which checks the NestedFolder feature on line: https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/permissions/dashboard.go#L208

File: pkg/infra/db/sqlbuilder.go [hunk 3]
```diff
@@ -69,6 +70,7 @@ func (sb *SQLBuilder) WriteDashboardPermissionFilter(user *user.SignedInUser, pe
 	)
 	if !ac.IsDisabled(sb.cfg) {
 		filterRBAC := permissions.NewAccessControlDashboardPermissionFilter(user, permission, "", sb.features, sb.recursiveQueriesAreSupported)
+		fmt.Printf("filterRBAC: %+v", filterRBAC)
```
Comment (IevaVasiljeva) [inline pkg/infra/db/sqlbuilder.go:73]: nit: debug log that should be removed.

File: pkg/services/alerting/store.go (hunk 1 above)
Comment (IevaVasiljeva) [reply pkg/services/alerting/store.go:172]: Yup, you're right. Do you mind creating a follow-up issue for this?

File: pkg/services/alerting/store.go [hunk 4]
```diff
@@ -179,6 +180,7 @@ func (ss *sqlStore) HandleAlertsQuery(ctx context.Context, query *alertmodels.Ge
 		}
 
 		alerts := make([]*alertmodels.AlertListItemDTO, 0)
+		fmt.Printf("%+v", builder.GetSQLString())
```
Comment (IevaVasiljeva) [inline pkg/services/alerting/store.go:183]: nit: again, a debug log.

**Review by IevaVasiljeva — COMMENTED** [review]
The changes look good to me! Happy for the PR to be merged after you remove the debug log and create an issue to remove the role check.

File: pkg/services/alerting/store.go (hunk 4 above)
Comment (eleijonmarck) [reply pkg/services/alerting/store.go:183]: yea left this in for myself :D

File: pkg/services/alerting/store.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/alerting/store.go:172]: https://github.com/grafana/grafana-enterprise/issues/4994 follow up issue

**Review by IevaVasiljeva — APPROVED** [review]

**Pull request merged**